Sensu2

treydock opened the pull request:
A few of the types are non-functional but all have issues opened with sensu-go.

treydock committed:
Refactor sensu_check to support Sensu 2

treydock committed:
Add proxy_requests property to sensu_check that factors into presence or absence of proxy-requests

treydock committed:
Remove unnecessary sensuctl method

treydock committed:
Add sensu_api_validator type that can be used to verify API is able to accept requests like those used with sensuctl

treydock committed:
Add acceptance test for sensu_check

treydock committed:
Have sensu_check autorequire all sensu_api_validator resources

treydock committed:
Split url parameter in url_host and url_port so can be passed to sensu_api_validator

treydock committed:
Rewrite sensu_handler type/provider for sensu2

treydock committed:
Refactor acceptance tests to perform multi node tests so that a backend container is seperate from agent container

treydock committed:
Be explicit about namevar of new types

treydock committed:
Move common array property setup to PuppetX::Sensu::ArrayProperty
Cleanup and a bit more work on sensu_check type

treydock committed:
Remove state file if `sensuctl configure` fails
Fix unit tests for sensu::backend

treydock committed:
New types will auto require Exec[sensuctl_configure]

treydock committed:
Move required property validation to create as doing in type was causing 'puppet resource' queries to fail

treydock committed:
Move type validation back to type, checking for ensure=>present seems to fix issue with 'puppet resource'

treydock committed:
Add custom property to sensu_check and sensu_handler
Add some new properties to sensu_check to match latest sensu-go source
Make new type properties more generic and move common logic to PuppetX modules

treydock committed:
Add more validations to sensu_handler

treydock committed:
Make unit tests for new types a bit more generic so easier to add and modify properties all at once
Used let based definition of types to make tests a bit more reliable

treydock committed:
Add sensu_asset type

treydock committed:
Move some common properties and parameters to PuppetX::Sensu::Type and extend this module in sensu_check and sensu_handler

treydock committed:
Partially reverts commit bf00d10340568ef11686aa0785ef4b57f7757130.
Also remove custom property from sensu_handler and sensu_asset per https://github.com/sensu/sensu-go/issues/1451

treydock committed:
Add sensu_entity type/provider

treydock committed:
Add sensu_filter type/provider

treydock committed:
Rename sensu_check custom property to extended_attributes, which matches what's in documentation

treydock committed:
Add output_metric_format and output_metric_handlers to sensu_check

treydock committed:
Add sensu_hook type/provider

treydock committed:
Add sensu_environment type/provider

treydock committed:
Add sensu_organization type/provider

treydock committed:
Use `sensuctl create` to handle both create and update of sensu_check

treydock committed:
Add array_matching => all to recent array properties of sensu_check

treydock committed:
Avoid running list with --all-organizations for upcoming role and user types

treydock committed:
Add sensu_role type/provider

treydock committed:
Refactor sensu_mutator for sensu v2

treydock commented:
I was going to add the default `sensu_organization` and `sensu_environment` resources to `sensu::backend` or maybe a class called by `sensu::backend` so that if someone did purging the defaults wouldn't have to be defined to avoid losing them.  Same could be done for `sensu_role`.  Right now both the environment and role types don't work so didn't implement that manifest change.

treydock committed:
Move yumrepo into private class and ensure sensu::repo class is required by all packages

treydock committed:
Move repo_class to parameter and put value in module hiera data

treydock committed:
Use rspec-puppet-facts for unit tests

treydock committed:
Run unit tests against Facter 3.x and force facterdb lookup to use facter 3.8.0

treydock committed:
Ensure backend is ready before starting sensu-agent on backend host.  This resolves issue with acceptance tests where catalog would have to be applied twice which fails the tests.

treydock committed:
Add ensure => absent acceptance tests for all types

sensu_asset currently skipped: https://github.com/sensu/sensu-go/issues/988

treydock committed:
Remove code not used by sensu2

treydock committed:
Remove operatingsystems from metadata.json that are not supported
Remove travis-ci acceptance tests for operating systems not yet supported

treydock committed:
Update sensu_version fact for sensu2.

Windows support is best guess and likely won't work.

treydock committed:
Fix unit tests and lint

treydock committed:
Match travis-ci tests of Puppet with metadata.json

treydock committed:
Refactor how repos are defined

treydock committed:
Add support for Debian 9 and Ubuntu 16.04

ghoneycutt commented on manifests/repo.pp:
```diff
@@ -1,9 +1,23 @@
+        gpgkey          => 'https://packagecloud.io/sensu/nightly/gpgkey',
+        sslverify       => 1,
+        sslcacert       => '/etc/pki/tls/certs/ca-bundle.crt',
+        metadata_expire => 300,
```
before => Package

treydock replied:
The reason I didn't do that is because the packages have `require => Class['sensu::repo']` since using before would only catch something like `sensu-agent` and would not work for `sensu-backend` and `sensu-cli`.

ghoneycutt replied:
aha.. let's continue with requiring the class and contains then

ghoneycutt commented on spec/classes/repo_spec.rb:
```diff
@@ -18,6 +18,21 @@
       end
+      context 'sensu::repo apt', :if => facts['osfamily'] == 'Debian' do
+        it {
+          should contain_apt__source('sensu_beta').with({
```
don't we need apt in the .fixtures.yml then

treydock replied:
Was already in .fixutres.yml.  The repo specs are not actually running, I need to fix them tonight.  I think the fact keys end up being symbols in unit tests with rspec-puppet-facts, at least that's my theory.  Doesn't affect manifest code, just effects how facts are referenced in tests.

treydock committed:
Integer type supports negative integers

treydock committed:
Add sensu_silenced type

Currently suffering from bug here: https://github.com/sensu/sensu-go/issues/1630

treydock commented:
@ghoneycutt Acceptance tests now pass after recent bug fixes to sensu-go.  The test of deleting an asset is skipped in acceptance tests as that operation not yet supported by sensu-go.

treydock committed:
Add support for Debian 8 and Ubuntu 14.04

treydock committed:
Add Ubuntu 18.04 support

Custom facterdb facts used to force facts to load when unit tests use 3.8.0 facterdb values

ghoneycutt commented on Vagrantfile:
```diff
@@ -77,6 +77,15 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
     agent.vm.provision :shell, :inline => "facter --custom-dir=/vagrant/lib/facter sensu_version"
   end
 
+  config.vm.define "ubuntu1804-agent", autostart: false do |agent|
```
```
==> ubuntu1804-agent: Running provisioner: shell...
    ubuntu1804-agent: Running: inline script
    ubuntu1804-agent: /tmp/vagrant-shell: line 1: puppet: command not found
The SSH command responded with a non-zero exit status. Vagrant
assumes that this means the command failed. The output for this command
should be in the log above. Please read the output to determine what
went wrong.
```

treydock replied:
The docker based tests work.  I ran into several issues with Ubuntu 18.04 and vagrant / virtual box.  One such issue was ifup/ifdown not being present and breaking some of the workflows needed by vagrant.  The issue about puppet missing is something else, will look into it.

treydock replied:
Arg, I forgot to commit change to vagrant boxe provisioning that used puppet5.  Ubuntu 18.04 is only supported by very latest puppet 5.
